src/Illuminate/Database/Eloquent/Relations/HasOneThrough.php

```diff
@@ -78,7 +78,11 @@ public function addOneOfManySubQueryConstraints(Builder $query, $column = null,
     {
         $query->addSelect([$this->getQualifiedFirstKeyName()]);
 
-        $this->performJoin($query);
+        // We need to join subqueries that are not the innermost subquery
+        // which is joined in the CanBeOneOfMany::ofMany method.
+        if ($this->getOneOfManySubQuery() !== null) {
+            $this->performJoin($query);
+        }
```

Comment:
This is admittedly not very pretty, but is necessary unless the `Relation::addConstraints` signature is changed to allow passing the query instance in (see Additional Considerations section in PR description, and the first commit diff https://github.com/laravel/framework/pull/55168/commits/818ebee35ed2422bddfa80ba7ebc9e6fb716c38a for an example).

The root problem is that **all** subqueries need the join applied, but the innermost query is already being joined in the `CanBeOneOfMany::ofMany` method (the `addConstraints` call).